zone/bot_raid.cpp

```diff
@@ -0,0 +1,287 @@
+/*	EQEMu: Everquest Server Emulator
+	Copyright (C) 2001-2016 EQEMu Development Team (http://eqemulator.org)
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; version 2 of the License.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY except by those people which sell it, which
+	are required to give you total support for your newly bought product;
+	without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program; if not, write to the Free Software
+	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+*/
+
+#include "bot.h"
+#include "object.h"
+#include "raids.h"
+#include "doors.h"
+#include "quest_parser_collection.h"
+#include "../common/data_verification.h"
+
+extern volatile bool is_zone_loaded;
+extern bool Critical; 
+
+std::vector<RaidMember> Raid::GetRaidGroupMembers(uint32 gid) 
+{
+	std::vector<RaidMember> raid_group_members;
+	raid_group_members.clear();
+
+	for (int i = 0; i < MAX_RAID_MEMBERS; ++i)
+	{
+		if (members[i].member && members[i].GroupNumber == gid)
+		{
+			raid_group_members.push_back(members[i]);
+		}
+	}
+	return raid_group_members;
+}
+
+// Returns Bot members that are in the raid
+// passing in owner will return all Bots that have the same owner.
+std::vector<Bot*> Raid::GetRaidBotMembers(uint32 owner)
+{
+	std::vector<Bot*> raid_members_bots;
+	raid_members_bots.clear();
+
+	for (int i = 0; i < MAX_RAID_MEMBERS; i++) {
+		if (
+			members[i].member &&
+			members[i].member->IsBot()
+			) {
+			auto b_member = members[i].member->CastToBot();
+			if (owner && b_member->GetBotOwnerCharacterID() == owner) {
+				raid_members_bots.emplace_back(b_member);
+			} else if (!owner) {
+				raid_members_bots.emplace_back(b_member);
+			}
+		}
+	}
+
+	return raid_members_bots;
+}
+
+// Returns Bot members that are in the group specified
+// passing in owner will return only Bots that have the same owner.
+std::vector<Bot*> Raid::GetRaidGroupBotMembers(uint32 gid)
+{
+	std::vector<Bot*> raid_members_bots;
+	raid_members_bots.clear();
+
+	for (int i = 0; i < MAX_RAID_MEMBERS; i++) {
```

Comment:
You can range through the array versus doing the integer iteration

eg

```
for (auto & m : members) {
```

Accessing

```cpp
m.member->DoThing()
```